contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,159 @@
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleV2Interface.sol";
+
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+
+/**
+ * @title Optimistic Arbitrator
+ * @notice This contract enables assertions to be made to the Optimistic Oracle and assertions to be ratified by the UMA
+ *  Data Verification Mechanism. An assertion consists of using the requestPrice and proposePrice methods of the OptimisticOracle to
+ *  confirm or deny the response to a question. To ratify an assertion, an initial assertion must be made and disputed, such that the
+ *  question is scaled to the DVM, where UMA token holders vote on it.
+ * @dev This contract is stateless and solely encapsulates the DVM's functionality to leverage the Optimistic Arbitrator pattern.
+ * @dev The Optimistic Oracle's functions are called on behalf of the caller, hence this contract will neither hold nor receive funds
+ *  from these actions.
+ */
+contract OptimisticArbitrator {
+    using SafeERC20 for IERC20;
+
+    FinderInterface public immutable finder;
+
+    OptimisticOracleV2Interface public immutable oo;
+
+    IERC20 public immutable currency;
+
+    bytes32 public priceIdentifier = "YES_OR_NO_QUERY";
+
+    /**
+     * @notice Constructor.
+     * @param _finderAddress finder to use to get addresses of DVM contracts.
+     * @param _currency collateral token used to pay rewards and fees.
+     */
+    constructor(address _finderAddress, address _currency) {
+        finder = FinderInterface(_finderAddress);
+        currency = IERC20(_currency);
+        oo = OptimisticOracleV2Interface(finder.getImplementationAddress(OracleInterfaces.OptimisticOracleV2));
+    }
+
+    /**
+     * @notice Makes an assertion to the Optimistic Oracle.
+     * @dev The Optimistic Oracle price proposal is submitted on behalf of the caller, therefore only the caller,
+     *  and not the Optimist Arbitrator, will hold or receive funds.
+     * @param timestamp timestamp of the price being requested.
+     * @param ancillaryData ancillary data representing additional args being passed with the price request.
+     * @param proposedPrice price being proposed.
+     * @param reward reward offered to a successful proposer. Note: this can be 0, which could make sense if the caller
+     * expects to not being disputed.
+     * @param bond custom proposal bond to set for request. If set to 0, defaults to the final fee.
+     * @param liveness custom proposal liveness to set for request, if set to 0, defaults to the default liveness value.
+     */
+    function makeAssertion(
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        int256 proposedPrice,
+        uint256 reward,
+        uint256 bond,
+        uint64 liveness
+    ) public {
+        _pullAndApprove(reward + bond + _getStore().computeFinalFee(address(currency)).rawValue);
+        _makeAssertion(timestamp, ancillaryData, proposedPrice, reward, bond, liveness);
```

Comment:
Sorry, I promise I'm not trying to be annoying. I thought the point of this contract was to:

> show how to build the escalation game directly in your contract and then ONLY arbitrate bonds within the OO. you could extend your contract implementation with this, resulting in internal escalation game support.

However, it looks like we're doing the request + proposal to the OO before the dispute happens (so the escalation game isn't internal). It also appears as if a dispute could return a result, that could be accessed via this contract, so we're letting the OO to arbitrate the result and not just the bonds.

If the answer is that we have decided that we _don't_ want this contract to show the pattern of Across-style internal escalation games + the OO just arbitrating bonds, that's totally fine! I'm just trying to make sure I understand what we're going for.

Reply:
to be clear @mrice32 @md0x 's contract here is not meant to show what you describe. it's meant to show how to use the OO do do direct calls from an integration contract (`makeAssertion` method) OR how to use use the OO do make a direct DVM request (`ratifyAssertion`) in the simplest posible form. This gives you _stateless assertions_ in the first case, and the second case shows how to go directly to the DVM via the OO and just use the OO to arbitrate bonds (like Sherlock protocol does).

The case you are talking about with internal contract escalation game @md0x actually built before in this same PR and the code can be found now within a separate PR here: https://github.com/UMAprotocol/dev-quickstart/pull/14

Reply:
Thanks @mrice32 and sorry for the confusion,
We discussed with @smb2796 and @chrismaree that in this contract we wanted to dispense with the internal escalation game and use DVM completely for result and bonds arbitrage. Some users already reproduce this usage on their own so we wanted to formalise it with this pattern. 

Sorry if I wasn't clear from the beginning: as I developed the contract, together with the team we changed the use case and finished defining this contract.

However we also decided (yesterday during sprit planning) to finally make the other flavour as well since we have other users who want to implement the internal escalation game. This last one will go in another PR